aten/src/ATen/core/ivalue.cpp

```diff
@@ -249,6 +249,41 @@ void IValue::dump() const {
   std::cout << *this << "\n";
 }
 
+IValue IValue::deepcopy() const {
```

Comment:
This does not seem to handle shared pointers to the same objects correctly. See the memo table in the python implementation of deep copy.